bindings/pyroot/pythonizations/python/ROOT/__init__.py

```diff
@@ -60,7 +60,70 @@
 import sys
 from ._facade import ROOTFacade
 
-sys.modules[__name__] = ROOTFacade(sys.modules[__name__], _is_ipython)
+_root_facade = ROOTFacade(sys.modules[__name__], _is_ipython)
+sys.modules[__name__] = _root_facade
+
+# Configure meta-path finder for ROOT namespaces, following the Python
+# documentation and an example:
+#
+#   * https://docs.python.org/3/library/importlib.html#module-importlib.abc
+#
+#   * https://python.plainenglish.io/metapathfinders-or-how-to-change-python-import-behavior-a1cf3b5a13ec
+from importlib.abc import Loader, MetaPathFinder
+from importlib.machinery import ModuleSpec
+from importlib.util import spec_from_loader
+
+
+def _lookup_root_module(fullname: str):
```

Comment:
```suggestion
def _lookup_root_module(fullname: str) -> Optional[types.ModuleType]:
```

Reply:
Or in general I would try to find a way to represent the type returned by this function

Reply:
I found a way to have better defined types by only returning `non-None` if the attribute is actually a Python module or a C++ namespace.